### PR TITLE
Add MAPE metric

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*
- Added MAPE metric (https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_absolute_percentage_error.html)
- Refactored eval_metric code to be cleaner and provide more helpful error messages communicating which metrics are supported in which problem types.
- Added unit test in LightGBM to test all eval metrics for each problem type to ensure they work without error.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1805-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1805/1/index.html

**Review by gradientsky — APPROVED** [review]

**Comment by gradientsky** [comment]
`__init__.py` has a lot of code inside. If tests has to be added, this might be challenging. Please consider moving all the logic outside of this file.
